test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -879,6 +879,30 @@ TEST_P(SslSocketTest, GetUriWithUriSan) {
                .setExpectedSerialNumber(TEST_SAN_URI_CERT_SERIAL));
 }
 
+TEST_P(SslSocketTest, IpSan) {
+  const std::string client_ctx_yaml =
+      fmt::format(R"EOF(
+  common_tls_context:
+    validation_context:
+      trusted_ca:
+        filename: "{{{{ test_rundir }}}}/test/config/integration/certs/upstreamcacert.pem"
```

Comment:
can you copy the cert to `test/extensions/transport_sockets/tls/test_data`?

Reply:
We need the CA cert also. Why copy it? What's the difference?

Reply:
erh, actually generate new cert with IP SAN only there with the CA there.

Reply:
What is the point of duplicating these certs in multiple places? I have the certs already that I need for this test? Why does it matter if they come from two places? If you really insist that I duplicate this I will do it next week.

Reply:
Sorry my initial intent is to have a test data purely for this unit test, i.e. with IP SAN only in the cert. Not really strong opinion though, up to you.

Reply:
I will add a TODO to cleanup the test certs. We have a lot of duplication for no reason IMO and it would be better to just have a single place where we generate out test certs.